test/unit/tcti-swtpm.c

```diff
@@ -308,6 +308,9 @@ tcti_swtpm_init_fail_connect_test (void **state)
     will_return (__wrap_connect, -1);
     ret = Tss2_Tcti_Swtpm_Init (ctx, &tcti_size, "host=127.0.0.1,port=666");
     assert_int_equal (ret, TSS2_TCTI_RC_IO_ERROR);
+
+    free(((TSS2_TCTI_SWTPM_CONTEXT*)ctx)->conf_copy);
+    free(ctx);
```

Comment:
if it really matters and you want to guarantee your free gets called, you can setup the test with cmocka_unit_test_setup_teardown()
and the setup function set the void **s so *s points to the tcti that's calloc'd. Then the teardown will always get called and execute the free.

or you could just point the TCTI on the heap, a 4096 buffer should always be big enough, and just assert that size < sizeof(buf).

Reply:
In all other tests the ctx was passed via state and freed in the teardown method. Here ctx was allocated locally. So I thought it's OK to free it locally, to achieve running the test without failure if the address sanitizer is activated.